src/handlers/pricing-label.ts

```diff
@@ -0,0 +1,183 @@
+import { Context } from "../types/context";
+
+import { createLabel, listLabelsForRepo, addLabelToIssue, clearAllPriceLabelsOnIssue } from "../shared/label";
+import { Label } from "../types/github";
+import { labelAccessPermissionsCheck } from "../shared/permissions";
+import { setPrice } from "../shared/pricing";
+import { handleParentIssue, isParentIssue, sortLabelsByValue } from "./handle-parent-issue";
+import { AssistivePricingSettings } from "../types/plugin-input";
+import { isIssueLabelEvent } from "../types/typeguards";
+
+export async function onLabelChangeSetPricing(context: Context): Promise<void> {
+  if (!isIssueLabelEvent(context)) {
+    context.logger.debug("Not an issue event");
+    return;
+  }
+  const config = context.config;
+  const logger = context.logger;
+  const payload = context.payload;
+
+  const labels = payload.issue.labels;
+  if (!labels) throw logger.error(`No labels to calculate price`);
+
+  if (payload.issue.body && isParentIssue(payload.issue.body)) {
+    await handleParentIssue(context, labels);
+    return;
+  }
+
+  const hasPermission = await labelAccessPermissionsCheck(context);
+  if (!hasPermission) {
+    if (config.publicAccessControl.setLabel === false) {
```

Comment:
This might need to be updated as we changed our config schema since the code was originally authored.